src/human_player_module.c

```diff
@@ -0,0 +1,391 @@
+ * Should be called only after validation of the user input
+ * 
+ * If user input is 3R, It will be mapped as follows
+ *  card.color=Red, card.name=3
+ * 
+ * @param input Value entered by the human during his turn
+ * 
+ * @return Card_t* Contains the mapped card name and color values
+ * If Card_t is NULL it means the input is an invalid one
+ */
+Card_t *map_user_input(char *input)
+{
+
+    Card_t *temp_card = (Card_t *)malloc(sizeof(Card_t));
+    temp_card = NULL;
+
+    if (input[0] == 'R')
+    {
+        temp_card->color = RED;
+    }
+    else if (input[0] == 'G')
+    {
+        temp_card->color = GREEN;
+    }
+    else if (input[0] == 'B')
+    {
+        temp_card->color = BLUE;
+    }
+
+    else if (input[0] == 'Y')
+    {
+        temp_card->color = YELLOW;
+    }
+    else
+    {
+        return temp_card;
+    }
+    //To map the number part of the User input to the card name field
+    temp_card->name = input[1] - '0';
+    return temp_card;
+}
+/**
+ * @brief Function picks a particular card from the current human deck of cards
+ * Called when a particular card has to be selected for removing from a deck
+ * Frees the menory allocated to he temporary selected drop card input parameter
+ * Note: Function  must be called only if the card to be selected is VALID
+ * 
+ * @param pp_head :Pointer which points to the address of head of card list in the Deck
+ * @param card_to_be_matched :Card which has to be matched from the deck
+ * 
+ * @return Card_t* Card that has been matched with the given card value
+ *         NULL - Returns NULL if the given card does not match with any of the cards in the deck
+ */
+Card_t *pick_card_from_deck(Deck_t **pp_head, Card_t *card_to_be_matched)
+{
+    Deck_t *temp = *pp_head;
+    while (temp->next != NULL)
+    {
+        if (temp->card.color == card_to_be_matched->color && temp->card.name == card_to_be_matched->name)
+        {
+            return &temp->card;
+        }
+        temp = temp->next;
+    }
+    free(card_to_be_matched); //Check if it can be freed here
+    return NULL;
+}
+
+/**
+ * @brief Displays the card assigned to the human player on the console during Request card/
+ * Draw1 and Draw2 scenarios
+ * 
+ * @param assigned_card Card that has been added newly to the player's deck
+ */
+void show_cards_assigned(Card_t assigned_card)
+{
+    printf("The card from the draw pile is %s%s \n", CARD_COLOR_STRING[assigned_card.color], CARD_NAME_STRING[assigned_card.name]);
+    display_player_deck(human_player);
+    return;
+}
+
+/**
+ * @brief Checks if the card dropped by human player during his turn is valid
+ *
+ * @param previous_card - Card that was dropped previously in the pile
+ * @param current_card - Current card dropped by the human player
+ * 
+ * @return- True if the card is invaid
+ *          False if the card is valid
+ */
+bool check_is_valid_turn(Card_t *previous_card, Card_t *current_card)
```

Comment:
No need to add input parameter Card_t *previous_card here. Since g_card_on_table is global, you can use it directly in the function.